src/video/sdl2_v.cpp

```diff
@@ -623,6 +623,22 @@ void VideoDriver_SDL_Base::LoopOnce()
 
 void VideoDriver_SDL_Base::MainLoop()
 {
+	/* We smoketest if the backend supports actually moving the
+	 * mouse. There is really no other way to detect it, as even
+	 * if, say, X11 reports it did it, X11-servers like remote
+	 * desktops or WSLg don't actually execute it. So the best
+	 * way to learn is to try. */
+	if (_settings_client.gui.scroll_mode == VSM_VIEWPORT_RMB_FIXED || _settings_client.gui.scroll_mode == VSM_MAP_RMB_FIXED) {
+		int x_before, x_after, y;
+		SDL_GetMouseState(&x_before, &y);
```

Comment:
SDL_GetMouseState always returns (0,0) for me on native X11.

If using SDL_GetGlobalMouseState and SDL_WarpMouseGlobal instead, it works.